src/java.base/share/classes/java/nio/charset/Charset.java

```diff
@@ -566,8 +566,7 @@ public static Charset forName(String charsetName,
     private static void put(Iterator<Charset> i, Map<String,Charset> m) {
         while (i.hasNext()) {
             Charset cs = i.next();
-            if (!m.containsKey(cs.name()))
-                m.put(cs.name(), cs);
+            m.putIfAbsent(cs.name(), cs);
```

Comment:
Yes, that would be a nicer. I've created JDK-8286366 for this. It could be changed to use for-each while you are there.

Reply:
> It could be changed to use for-each while you are there.

Not sure I get the idea.
Do you mean something like this:
```
        i.forEachRemaining(new Consumer<Charset>() {
            @Override
            public void accept(Charset cs) {
                m.putIfAbsent(cs.name(), cs);
            }
        });
```

Reply:
Sorry for the confusion, I thought I was looking at an Iterable at the time.